### PR TITLE
Add Code QL analysis

### PR DESCRIPTION
**Pull request by vasireddy99**
This PR  will add the CodeQL analysis, which helps to discover vulnerabilities across a codebase, with every change that is being pushed.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by qhanam** [comment]
@vasireddy99 please provide context for this PR in the description.
1. What is the problem being solved?
2. How does this change solve the problem?

**Comment by vasireddy99** [comment]
> @vasireddy99 please provide context for this PR in the description.
> 
> 1. What is the problem being solved?
> 2. How does this change solve the problem?

thanks @qhanam :-)

**Comment by qhanam** [comment]
Hi @vasireddy99, this still isn't enough information for us to make a decision on whether or not to integrate this static analysis tool into our build process.

I see this is a tool [developed by GitHub](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning-with-codeql). However, it is not clear to me what its behavior is, and what benefit we get from integrating it.

For example, once the database has been created, what do we do with it? Does GitHub automatically run queries against it to identify known or emerging vulnerabilities? Do we need to run queries against it ourselves? Which queries would we run, and why?

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
```
Comment (qhanam) [inline .github/workflows/codeql-analysis.yml:11]: issue: Please clean up this file by removing copied instructional comments such as these.

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        languages: ${{ matrix.language }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
+        # By default, queries listed here will override any specified in a config file.
+        # Prefix the list here with "+" to use these queries and those in the config file.
+        # queries: ./path/to/local/query, your-org/your-repo/queries@main
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
```
Comment (qhanam) [inline .github/workflows/codeql-analysis.yml:49]: issue: This is a TypeScript repo, I don't think this step applies.

Is there a reason why the NPM build would need to run? How does CodeQL handle JavaScript?

**Comment by vasireddy99** [comment]
Fixed the lint check in the PR

**Review by qhanam — APPROVED** [review]

**Pull request merged**